runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -46,6 +46,21 @@ IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
   }
 }
 
+IntervalSet::~IntervalSet()
+{
+}
+
+IntervalSet& IntervalSet::operator=(const IntervalSet& other)
+{
+  if (_readonly) {
+    throw IllegalStateException("can't alter read only IntervalSet");
+  }
```

Comment:
Thinking further, it seems to me that _readonly is a Java way of getting const references. `ATN::nextTokens()`, for example, returns a non-const reference to a cache IntervalSet, and there has to be a way to prevent callers from modifying the cache.

In this world-view, clearing the read only flag on copy seems correct. It also seems like the flag could be removed if the code started return const references rather than non-const references in that case. Making all this stuff const-correct seems like a big job, though.